packages/types/src/Keystore.ts

```diff
@@ -48,98 +49,40 @@ export interface ResponseData<A extends string> {
-   * @param signData.meta An optional partial [[SignerPayloadJSON]] object which can be used to communicate information about the data to be signed to the Keystore.
-   * @returns A Promise resolving to [[ResponseData]] containing the signature or rejecting if key or algorithm is unknown or if they do not match.
-   */
-  sign<A extends SignAlgs>(
-    signData: KeystoreSigningData<A>
-  ): Promise<ResponseData<A>>
-  /**
-   * Uses stored key material to encrypt a message encoded as u8a.
-   *
-   * @param requestData [[RequestData]] containing the data to be encrypted, the algorithm to be used, and the public key identifying the key to use.
-   * @returns A Promise resolving to [[ResponseData]] containing the encrypted data or rejecting if key or algorithm is unknown or if they do not match.
-   */
-  encrypt<A extends EncryptAlgs>(
-    requestData: RequestData<A>
-  ): Promise<ResponseData<A>>
-  /**
-   * Uses stored key material to decrypt a message encoded as u8a.
-   *
-   * @param requestData [[RequestData]] containing the message to be decrypted, the algorithm to be used, and the public key identifying the key to use.
-   * @returns A Promise resolving to [[ResponseData]] containing the decrypted message or rejecting if key or algorithm is unknown or if they do not match.
-   */
-  decrypt<A extends EncryptAlgs>(
-    requestData: RequestData<A>
-  ): Promise<ResponseData<A>>
-  /**
-   * Can be used to check whether a given key is stored for a given algorithm.
-   *
-   * @param keys Array of objects containing an algorithm identifier and public key.
-   * @returns Array of boolean values indicating if a key is known or not.
-   */
-  hasKeys(
-    keys: Array<Pick<RequestData<string>, 'alg' | 'publicKey'>>
-  ): Promise<boolean[]>
-  /**
-   * Returns an array containing information on each key pair stored by the keystore.
-   *
-   * @returns An array of items { alg: string, publicKey: Uint8Array }.
-   */
-  getKeys?(): Promise<Array<Pick<RequestData<string>, 'alg' | 'publicKey'>>>
-}
+export type SignCallback<A extends string = any> = (
+  signData: SigningData<A>
+) => Promise<ResponseData<A>>
 
-/**
- * A Keystore instance which supports signing with a given signature algorithm.
- */
-export type KeystoreSigner<A extends string = any> = Pick<
-  Keystore<A, string>,
-  'sign'
->
-
-export interface KeystoreSigningOptions<A extends string = any> {
-  signer: KeystoreSigner<A>
+export interface SigningOptions<A extends string = any> {
+  sign: SignCallback<A>
   signingPublicKey: string | Uint8Array
   alg: A
 }
 
 /**
- * A Keystore instance which supports {en/de}cryption using nacl's 'x25519-xsalsa20-poly1305' algorithm.
+ * Methods for {en/de}cryption using nacl's 'x25519-xsalsa20-poly1305' algorithm.
  */
-export interface NaclBoxCapable
-  extends Keystore<any, 'x25519-xsalsa20-poly1305'> {
+export interface NaclBoxCapable {
```

Comment:
Should we extract the `encrypt` and `decrypt` properties here into two separate callbacks as we did with the SigningCallback, too?

Reply:
Yes, I was considering this, will do!